python/python/lance/sampler.py

```diff
@@ -143,7 +143,7 @@ def _filtered_efficient_sample(
             limit=(end - start),
             batch_size=shard_size,
         )
-        if len(columns) == 1 and filter.lower() == f"{columns[0]} is not null":
+        if len(columns) == 1 and filter.lower() == f"{columns[0]}.lower() is not null":
```

Comment:
Hi, it looks like we are comparing with `NAME.lower() is not null`, should we inlcude `.lower()` in the `{columns[0].lower()}`?